langchain/llms/huggingface_pipeline.py

```diff
@@ -0,0 +1,106 @@
+"""Wrapper around HuggingFace Pipeline APIs."""
+from typing import Any, Dict, List, Mapping, Optional
+
+from pydantic import BaseModel, Extra, root_validator
+
+from langchain.llms.base import LLM
+from langchain.llms.utils import enforce_stop_tokens
+
+DEFAULT_MODEL_ID = "gpt2"
+DEFAULT_TASK = "text-generation"
+VALID_TASKS = ("text2text-generation", "text-generation")
+
+
+class HuggingFacePipeline(LLM, BaseModel):
+    """Wrapper around HuggingFace Pipeline API.
+
+    To use, you should have the ``transformers`` python package installed.
+
+    Only supports `text-generation` and `text2text-generation` for now.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.llms.huggingface_pipeline import HuggingFacePipeline
+            hf = HuggingFacePipeline(model_id="gpt2", task="text-generation")
+    """
+
+    pipeline: Any  #: :meta private:
+    model_id: str = DEFAULT_MODEL_ID
+    """Model name to use."""
+    task: str = DEFAULT_TASK
+    """Task to call the model with. Should be a task that returns `generated_text`."""
+    model_kwargs: Optional[dict] = None
+    """Key word arguments to pass to the model."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate python package exists in environment."""
+        try:
+            if values.get("pipeline") is None:
+                values["pipeline"] = cls.from_model_id(values)
+
+        except ImportError:
+            raise ValueError(
+                "Could not import transformers python package. "
+                "Please it install it with `pip install transformers`."
+            )
+
+        return values
+
+    @staticmethod
+    def from_model_id(values):
```

Comment:
i would do something like
```
@classmethod
def from_model_id(model_id: str, task: Optional[str] = None, model_kwargs: Optional[dict] = None, **kwargs):
    # construct pipeline here
    return cls(pipeline=pipeline, **kwargs)
```

this allows users to do either

`HuggingFacePipeline(pipeline=pipeline)` or `HuggingFacePipeline.from_model_id("gpt2")`

i think if we do this, we can also just delete the above validate_environment call

i think we also no longer need to save the model_id, model_kwargs, and task param on the class (which simplifies the interface imensely)

Reply:
_identifying_params probably becomes something just based on the pipeline - what would that look like? we use that for caching...

Reply:
Moved to classmethod. Will think about changing identifying params now

Reply:
makes sense, i think thats reasonable